lib/rubygems/specification.rb

```diff
@@ -726,6 +719,21 @@ def test_files=(files) # :nodoc:
     @test_files = Array files
   end
 
+  ######################################################################
+  # :section: Read-only attributes
+
+  ##
+  # The version of RubyGems used to create this gem.
+
+  attr_accessor :rubygems_version
+
+  ##
+  # The path where this gem installs its extensions.
+
+  def extensions_dir
+    @extensions_dir ||= super
```

Comment:
Any reason to not use just `super` and make it clear in comment the only purpose of this re-definition is spec reference completeness. It should be possible to avoid this commend in reference guide using `#--` directive (https://ruby.github.io/rdoc/RDoc/Markup.html#class-RDoc::Markup-label-Documenting+Source+Code).

Reply:
There was no strong reason, I just thought that memoizing would not heart, but your idea makes this patch less invasive and more focused on what we want to achieve. I will update it as suggested, thanks!

Reply:
I'm trying to be more conservative to not introduce additional problems for extensions now. :pray: